api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -388,6 +398,29 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
     dbWorkspace.addWorkspaceUserRole(permissions);
 
     dbWorkspace = workspaceService.getDao().save(dbWorkspace);
+
+    JSONObject demoCohort = cloudStorageService.readDemoCohort();
+    // TODO: Make this a service method to avoid duplication with CohortsController
+    Cohort dbCohort = new Cohort();
+    dbCohort.setName(demoCohort.getString("name"));
+    dbCohort.setDescription(demoCohort.getString("description"));
+    dbCohort.setCriteria(demoCohort.get("criteria").toString());
+    dbCohort.setType("AoU_Discover");
+    dbCohort.setCreator(userProvider.get());
+    dbCohort.setWorkspaceId(dbWorkspace.getWorkspaceId());
+    dbCohort.setCreationTime(now);
+    dbCohort.setLastModifiedTime(now);
+    dbCohort.setVersion(1);
+    try {
+      // TODO Make this a pre-check within a transaction?
+      dbCohort = cohortDao.save(dbCohort);
+    } catch (DataIntegrityViolationException e) {
+      // TODO The exception message doesn't show up anywhere; neither logged nor returned to the
+      // client by Spring (the client gets a default reason string).
+      throw new BadRequestException(String.format(
```

Comment:
Hmm.. do we actually need to handle this case specifically? I'd guess there are many other exceptions that are more likely than this. The explanation given here implies someone else inserted a cohort with a conflicting column, a few milliseconds after we created the workspace. Did you observe this exception?

Reply:
Similarly copied from the cohorts controller. I did, however, get it once when I made two demo cohort files (Just pushed. Should have let you know more work ended up needing to happen), so that was useful for my development of demo files at least!